lib/Controller/VideoController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * @copyright Copyright (c) 2021 Frederic Ruget <fred@atlant.is>
+ *
+ * @author Frederic Ruget (douzebis) <fred@atlant.is>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Viewer\Controller;
+
+use Exception;
+
+use OCP\IRequest;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\Controller;
+use OCP\L10N\IFactory;
+
+use OCP\Files\File;
+use OCP\Files\IRootFolder;
+use OCP\IUserSession;
+
+class VideoController extends Controller {
+
+	// I copied the annotation below without understanding it... is it useful?
+	/** @var IRootFolder */
+	private $storage;
+
+	// I copied the annotation below without understanding it... is it useful?
+	/** @var array */
+	private $locales;
+
+	public function __construct(
+		string $AppName,
+		IRequest $request,
+		IRootFolder $storage,
+		IFactory $factory,
+		IUserSession $userSession
+	){
+		parent::__construct($AppName, $request);
+		$this->storage = $storage;
+		$this->locales = $factory->findAvailableLocales();;
+		$this->currentUser = $userSession->getUser();
```

Comment:
We usually put only services and do the callings on the methods.

```suggestion
		$this->rootFolder = $rootFolder;
		$this->l10nFactory = $l10nFactory;
		$this->userSession = $userSession;
```

Reply:
Also, you're missing the private properties for those :wink:

Reply:
Hmm...
Meaning `rootFolder`, `l10nFactory`, `userSession` should be private properties of the `VideoController`class ?
E.g.: `private $rootFolder`?

Reply:
yes, see other examples of those controllers
https://github.com/nextcloud/server/blob/d89a75be0b01f0423a7c1ad2d58aac73c3cc1f3a/apps/dav/lib/Controller/DirectController.php#L45-L80